clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/BasicAmazonDeployHandler.groovy

```diff
@@ -265,6 +265,8 @@ class BasicAmazonDeployHandler implements DeployHandler<BasicAmazonDeployDescrip
       copyScalingPoliciesAndScheduledActions(
         task, sourceRegionScopedProvider, description.credentials, description.source.asgName, region, asgName
       )
+
+      createLifecycleHooks(task, regionScopedProvider, account, asgName)
```

Comment:
I've updated the codebase to support both adding deploy-time lifecycle hooks, as well as optionally opt-ing out of account lifecycle hooks.